src/sys/select.rs

```diff
@@ -24,37 +27,40 @@ fn assert_fd_valid(fd: RawFd) {
     );
 }
 
-impl FdSet {
+impl<'fd> FdSet<'fd> {
     /// Create an empty `FdSet`
-    pub fn new() -> FdSet {
+    pub fn new() -> FdSet<'fd> {
         let mut fdset = mem::MaybeUninit::uninit();
         unsafe {
             libc::FD_ZERO(fdset.as_mut_ptr());
-            FdSet(fdset.assume_init())
+            Self {
+                set: fdset.assume_init(),
+                _fd: std::marker::PhantomData,
+            }
         }
     }
 
     /// Add a file descriptor to an `FdSet`
-    pub fn insert(&mut self, fd: RawFd) {
-        assert_fd_valid(fd);
-        unsafe { libc::FD_SET(fd, &mut self.0) };
+    pub fn insert<Fd: AsFd>(&mut self, fd: &'fd Fd) {
+        assert_fd_valid(fd.as_fd().as_raw_fd());
+        unsafe { libc::FD_SET(fd.as_fd().as_raw_fd(), &mut self.set) };
     }
 
     /// Remove a file descriptor from an `FdSet`
-    pub fn remove(&mut self, fd: RawFd) {
-        assert_fd_valid(fd);
-        unsafe { libc::FD_CLR(fd, &mut self.0) };
+    pub fn remove<Fd: AsFd>(&mut self, fd: Fd) {
```

Comment:
For `FdSet::remove()`, IMO, it is fine to "consume" that `fd` as it will no long be monitored.

Reply:
I disagree.  It's conceivable that somebody no longer wishes to poll a certain file descriptor with select, but still wishes to have it open.  That's harder to do in this case.  Plus, for consistency's sake, I think remove should take by reference just like the other methods.

Reply:
Done.